Tools/Scripts/libraries/webkitbugspy/webkitbugspy/bugzilla.py

```diff
@@ -112,7 +112,9 @@ def from_string(self, string):
 
     def credentials(self, required=True, validate=False):
         def validater(username, password):
-            response = requests.get('{}/rest/user/{}?login={}&password={}'.format(self.url, username, username, password))
+            quoted_username = requests.utils.quote(username)
```

Comment:
`test-webkitpy` tests are failing. Need to change `CREDENTIAL_RE` in `Tools/Scripts/libraries/webkitbugspy/webkitbugspy/mocks/bugzilla.py` too.

Reply:
Oh no, not a regex :(

Reply:
It's just used for testing, so if you pass the unit tests, you're good!

Reply:
OK I've entered the test-webkitpy rabbit hole.

I will deal with https://bugs.webkit.org/show_bug.cgi?id=239786.

Maybe you could look at https://bugs.webkit.org/show_bug.cgi?id=239788?

Reply:
I will look at that one...that's probably just laziness when writing the unit test on my part.

Reply:
> Need to change CREDENTIAL_RE in Tools/Scripts/libraries/webkitbugspy/webkitbugspy/mocks/bugzilla.py too.

Are you sure something is wrong with that regex? I see a bunch of tests are failing, but it doesn't look related to the regex.

Reply:
If you look at https://ews-build.webkit.org/#/builders/5/builds/78706, something like `webkitbugspy.tests.bugzilla_unittest.TestBugzilla.test_create` is definitely failing because you've changed what the requests to bugzilla look like.

That's all `CREDENTIAL_RE` is doing, by the way, we're matching a string in the request in our mock server to extract the username and password.

Reply:
It was actually failing _before_ checking `CREDENTIAL_RE`.

Anyway, you suggested a better solution for resolving this -- URL decode in the mock implementation -- and that worked.